contracts/TokenFactory.sol

```diff
@@ -305,64 +305,35 @@ contract TokenFactory is ProxyFactory, Ownable, ITokenFactory {
         return false;
     }
 
-    function mint721(
+    function mintBatch721(
         address nft,
         address to,
-        uint256 tokenId,
-        bytes memory data,
+        uint256[] calldata tokenIds,
+        bytes calldata data,
         uint8 v,
         bytes32 r,
         bytes32 s
-    ) public override {
+    ) external override {
         address owner = IBaseNFT721(nft).owner();
-        bytes32 hash = keccak256(abi.encode(NFT721_TYPEHASH, nft, to, tokenId, data, nonces721[owner]++));
+        bytes32 hash = keccak256(abi.encode(NFT721_TYPEHASH, nft, to, tokenIds, data, nonces721[owner]++));
         Signature.verify(hash, owner, v, r, s, DOMAIN_SEPARATOR());
-        IBaseNFT721(nft).mint(to, tokenId, data);
+        IBaseNFT721(nft).mintBatch(to, tokenIds, data);
```

Comment:
https://github.com/sushiswap/shoyu/blob/master/contracts/base/BaseNFT721.sol#L121
https://github.com/sushiswap/shoyu/blob/master/contracts/base/BaseNFT721.sol#L131

The require conditions should be swapped.